File: backend/spring-boot/src/main/java/org/bootstrapbugz/api/user/service/ProfileService.java
package org.bootstrapbugz.api.user.service;

import org.bootstrapbugz.api.user.payload.dto.UserDTO;
import org.bootstrapbugz.api.user.payload.request.ChangePasswordRequest;
import org.bootstrapbugz.api.user.payload.request.PatchProfileRequest;

public interface ProfileService {
  UserDTO find();

  UserDTO patch(PatchProfileRequest patchProfileRequest);

  void changePassword(ChangePasswordRequest changePasswordRequest);

  void delete();
}

File: backend/spring-boot-kotlin/src/main/java/org/bootstrapbugz/api/user/service/ProfileService.kt
package org.bootstrapbugz.api.user.service

import org.bootstrapbugz.api.user.payload.dto.UserDTO
import org.bootstrapbugz.api.user.payload.request.ChangePasswordRequest
import org.bootstrapbugz.api.user.payload.request.PatchProfileRequest

interface ProfileService {
    fun find(): UserDTO
    fun patch(patchProfileRequest: PatchProfileRequest): UserDTO
    fun delete()
    fun changePassword(changePasswordRequest: ChangePasswordRequest)
}
